src/pages/signin/PasswordForm.js

```diff
@@ -52,202 +52,213 @@ const defaultProps = {
+    const validateAndSubmitForm = useCallback(() => {
+        const passwordTrimmed = password.trim();
+        const twoFactorCodeTrimmed = twoFactorAuthCode.trim();
+        const requiresTwoFactorAuth = props.account.requiresTwoFactorAuth;
+
+        if (!passwordTrimmed) {
+            setFormError({password: 'passwordForm.pleaseFillPassword'});
+            return;
+        }
+
+        if (!ValidationUtils.isValidPassword(passwordTrimmed)) {
+            setFormError({password: 'passwordForm.error.incorrectPassword'});
+            return;
+        }
 
-        if (!password) {
-            this.setState({formError: {password: 'passwordForm.pleaseFillPassword'}});
+        if (requiresTwoFactorAuth && !twoFactorCodeTrimmed) {
+            setFormError({twoFactorAuthCode: 'passwordForm.pleaseFillTwoFactorAuth'});
             return;
         }
 
-        if (!ValidationUtils.isValidPassword(password)) {
-            this.setState({formError: {password: 'passwordForm.error.incorrectPassword'}});
+        if (requiresTwoFactorAuth && !ValidationUtils.isValidTwoFactorCode(twoFactorCodeTrimmed)) {
+            setFormError({twoFactorAuthCode: 'passwordForm.error.incorrect2fa'});
             return;
         }
 
-        if (requiresTwoFactorAuth && !twoFactorCode) {
-            this.setState({formError: {twoFactorAuthCode: 'passwordForm.pleaseFillTwoFactorAuth'}});
+        setFormError({});
+
+        Session.signIn(passwordTrimmed, '', twoFactorCodeTrimmed, props.preferredLocale);
+    }, [password, twoFactorAuthCode, props.account.requiresTwoFactorAuth, props.preferredLocale]);
+
+    useEffect(() => {
+        if (!canFocusInputOnScreenFocus() || !inputPasswordRef.current || !props.isVisible) {
+            return;
+        }
+        inputPasswordRef.current.focus();
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- we don't want this effect to run again
+    }, []);
+
+    useEffect(() => {
+        if (props.isVisible) {
+            inputPasswordRef.current.focus();
+        }
+        if (props.isVisible && password) {
+            clearPassword();
+        }
+        // We cannot add password to the dependency list because it will clear every time it updates
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [props.isVisible]);
+
+    useEffect(() => {
+        if (!props.account.requiresTwoFactorAuth) {
             return;
         }
+        input2FA.current.focus();
+    }, [props.account.requiresTwoFactorAuth]);
 
-        if (requiresTwoFactorAuth && !ValidationUtils.isValidTwoFactorCode(twoFactorCode)) {
-            this.setState({formError: {twoFactorAuthCode: 'passwordForm.error.incorrect2fa'}});
+    useEffect(() => {
+        if (twoFactorAuthCode.length !== CONST.TFA_CODE_LENGTH) {
             return;
         }
+        validateAndSubmitForm();
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- We don't need to call this when the function changes.
```

Comment:
Well it was more of an optimisation tweak to make sure that we are not running this on any unintentional changes, but I don't think it will matter if we include it or not. So anyway, I'll include it.